report/generator/file-namer.js

```diff
@@ -35,11 +35,16 @@ function getFilenamePrefix(name, fetchTime) {
 
 /**
  * Generate a filenamePrefix of hostname_YYYY-MM-DD_HH-MM-SS.
- * @param {{finalUrl: string, fetchTime: string}} lhr
+ * @param {{finalDisplayedUrl: string, finalUrl?: string, fetchTime: string}} lhr
  * @return {string}
  */
 function getLhrFilenamePrefix(lhr) {
-  const hostname = new URL(lhr.finalUrl).hostname;
+  // If LHR is older than 10.0 it will not have the `finalDisplayedUrl` property.
+  // Old LHRs should have the `finalUrl` property which will work fine for the report.
+  let url = lhr.finalDisplayedUrl;
+  if (!url && lhr.finalUrl) url = lhr.finalUrl;
```

Comment:
Repeating code isn't ideal here, but this file is in it's own CJS world so it seemed better than refactoring that

Reply:
Not sure I follow, but if there is something here that should change when converted to es modules please leave a `TODO(esmodules)` comment.

Reply:
I think it has less to do with ESM and more to do with this being a shared dependency. Still might need a TODO that addresses the shared dep stuff with this file.